.github/workflows/rust.yml

```diff
@@ -19,102 +19,39 @@ jobs:
           # 32-bit target
           - target: i686-unknown-linux-gnu
             deps: sudo apt update && sudo apt install gcc-multilib
-
           # 64-bit target
           - target: x86_64-unknown-linux-gnu
     steps:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@stable
       - run: rustup target add ${{ matrix.target }}
       - run: ${{ matrix.deps }}
+      - run: cargo test --target ${{ matrix.target }} --no-default-features --features alloc
       - run: cargo test --target ${{ matrix.target }}
+      - run: cargo test --target ${{ matrix.target }} --features batch
+      - run: cargo test --target ${{ matrix.target }} --features batch_deterministic
+      - run: cargo test --target ${{ matrix.target }} --features serde
 
   test-simd:
     name: Test simd backend (nightly)
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions-rs/toolchain@v1
-      with:
-        profile: minimal
-        toolchain: nightly
-        override: true
-    - uses: actions-rs/cargo@v1
-      with:
-        command: test
-        args: --no-default-features --features "std nightly simd_backend"
-
-  test-defaults-serde:
-    name: Test default feature selection and serde
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-    - uses: actions-rs/toolchain@v1
-      with:
-        profile: minimal
-        toolchain: stable
-        override: true
-    - uses: actions-rs/cargo@v1
-      with:
-        command: test
-        args: --features "serde"
-
-  test-alloc-u32:
-    name: Test no_std+alloc
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-    - uses: actions-rs/toolchain@v1
-      with:
-        profile: minimal
-        toolchain: stable
-        override: true
-    - uses: actions-rs/cargo@v1
-      with:
-        command: test
-        args: --lib --no-default-features --features "alloc"
-
-  test-batch-deterministic:
-    name: Test deterministic batch verification
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-    - uses: actions-rs/toolchain@v1
-      with:
-        profile: minimal
-        toolchain: stable
-        override: true
-    - uses: actions-rs/cargo@v1
-      with:
-        command: test
-        args: --features "batch_deterministic"
+      - uses: actions/checkout@v2
+      - uses: dtolnay/rust-toolchain@nightly
+      - run: cargo test --features simd_backend
 
   msrv:
```

Comment:
Could you make this equivalent to the [more thorough](https://github.com/dalek-cryptography/curve25519-dalek/blob/release/4.0/.github/workflows/rust.yml#L72) MSRV build test from curve25519-dalek?